src/Tizen.Network.Nsd/Tizen.Network.Nsd/DnssdService.cs

```diff
@@ -217,15 +219,89 @@ public IPAddressInformation IP
                 return IPAddressInstance;
             }
         }
+        
+        private void printTxt(byte [] value)
+        {
+            var sb = new StringBuilder("new byte[] { ");
+            foreach (var bt in value)
+            {
+                sb.Append(Convert.ToChar(bt));
+            }
+            sb.Append("}");
+            Log.Debug(Globals.LogTag, "PPKK2 txtRecord: " + sb.ToString());
+        }
+        
+        /// <summary>
+        /// Gets all TXT records in the form of key and value.
+        /// </summary>
+        /// <remarks>
+        /// After adding the TXT record call for get TXT record.
+        /// </remarks>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name="map">gets the each record key, value in the map</param>
+        /// <feature>http://tizen.org/feature/network.service_discovery.dnssd</feature>
+        /// <exception cref="NotSupportedException">Thrown when DNS-SD is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when any other error occurred.</exception>
+        public void GetTxtRecord(Dictionary<string, string> map)
```

Comment:
How about create new dictionary in this method and return IDictionary<string, string> type instead of using reference object of map?